thingiftest/src/androidTest/java/com/kii/thingif/largetests/TriggerTest.java

```diff
@@ -0,0 +1,781 @@
+package com.kii.thingif.largetests;
+
+import android.support.test.runner.AndroidJUnit4;
+import android.text.TextUtils;
+import android.util.Pair;
+
+import com.google.gson.JsonObject;
+import com.kii.cloud.rest.client.KiiRest;
+import com.kii.cloud.rest.client.model.KiiCredentials;
+import com.kii.cloud.rest.client.model.storage.KiiThing;
+import com.kii.thingif.OnboardWithVendorThingIDOptions;
+import com.kii.thingif.Target;
+import com.kii.thingif.ThingIFAPI;
+import com.kii.thingif.TypedID;
+import com.kii.thingif.actions.AirConditionerActions;
+import com.kii.thingif.actions.HumidityActions;
+import com.kii.thingif.clause.trigger.EqualsClauseInTrigger;
+import com.kii.thingif.clause.trigger.RangeClauseInTrigger;
+import com.kii.thingif.command.Action;
+import com.kii.thingif.command.AliasAction;
+import com.kii.thingif.command.Command;
+import com.kii.thingif.exception.BadRequestException;
+import com.kii.thingif.trigger.Condition;
+import com.kii.thingif.trigger.EventSource;
+import com.kii.thingif.trigger.ScheduleOncePredicate;
+import com.kii.thingif.trigger.SchedulePredicate;
+import com.kii.thingif.trigger.ServerCode;
+import com.kii.thingif.trigger.StatePredicate;
+import com.kii.thingif.trigger.Trigger;
+import com.kii.thingif.trigger.TriggerOptions;
+import com.kii.thingif.trigger.TriggeredCommandForm;
+import com.kii.thingif.trigger.TriggeredServerCodeResult;
+import com.kii.thingif.trigger.TriggersWhen;
+
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RunWith(AndroidJUnit4.class)
+public class TriggerTest extends LargeTestCaseBase{
+    @Test
+    public void basicTriggerTest() throws Exception {
+        ThingIFAPI api = this.createDefaultThingIFAPI();
+        String vendorThingID = UUID.randomUUID().toString();
+        String thingPassword = "password";
+
+        // on-boarding thing
+        OnboardWithVendorThingIDOptions options =
+                new OnboardWithVendorThingIDOptions.Builder()
+                        .setThingType(DEFAULT_THING_TYPE)
+                        .setFirmwareVersion(DEFAULT_FIRMWARE_VERSION).build();
+        Target target = api.onboardWithVendorThingID(vendorThingID, thingPassword, options);
+        Assert.assertEquals(TypedID.Types.THING, target.getTypedID().getType());
+        Assert.assertNotNull(target.getAccessToken());
```

Comment:
How about creating `ThingIFAPI` instance and onboarding the instance in `setup()` method?